lib/reversetunnel/fake.go

```diff
@@ -51,12 +51,25 @@ type FakeRemoteSite struct {
 	RemoteSite
 	// Name is the remote site name.
 	Name string
-	// ConnCh receives the connection when dialing this site.
-	ConnCh chan net.Conn
 	// AccessPoint is the auth server client.
 	AccessPoint auth.RemoteProxyAccessPoint
 	// OfflineTunnels is a list of server IDs that will return connection error.
 	OfflineTunnels map[string]struct{}
+	// connCh receives the connection when dialing this site.
+	connCh chan net.Conn
+	// closedMtx is a mutex that protects closed.
+	closedMtx sync.Mutex
+	// closed is set to true after the site is being closed.
+	closed bool
```

Comment:
nit: the closedMtx call be replaced by atomic.Load and atomic.StoreInt operations.